src/AppInstallerRepositoryCore/Rest/Schema/1_5/Json/ManifestDeserializer_1_5.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "ManifestDeserializer.h"
+#include <winget/JsonUtil.h>
+
+using namespace AppInstaller::Manifest;
+
+namespace AppInstaller::Repository::Rest::Schema::V1_5::Json
+{
+    namespace
+    {
+        // Locale
+        constexpr std::string_view Icons = "Icons"sv;
+        constexpr std::string_view IconUrl = "IconUrl"sv;
+        constexpr std::string_view IconFileType = "IconFileType"sv;
+        constexpr std::string_view IconResolution = "IconResolution"sv;
+        constexpr std::string_view IconTheme = "IconTheme"sv;
+        constexpr std::string_view IconSha256 = "IconSha256"sv;
+    }
+
+    std::optional<Manifest::ManifestLocalization> ManifestDeserializer::DeserializeLocale(const web::json::value& localeJsonObject) const
+    {
+        auto result = V1_4::Json::ManifestDeserializer::DeserializeLocale(localeJsonObject);
+
+        if (result)
+        {
+            auto& locale = result.value();
+
+            // Icons
+            auto iconsNode = JSON::GetRawJsonArrayFromJsonNode(localeJsonObject, JSON::GetUtilityString(Icons));
+            if (iconsNode)
+            {
+                std::vector<Manifest::Icon> icons;
+                for (auto const& iconNode : iconsNode->get())
+                {
+                    Manifest::Icon iconEntry;
+
+                    iconEntry.Url = JSON::GetRawStringValueFromJsonNode(iconNode, JSON::GetUtilityString(IconUrl)).value_or("");
+                    if (iconEntry.Url.empty())
+                    {
+                        continue;
+                    }
+
+                    iconEntry.FileType = Manifest::ConvertToIconFileTypeEnum(JSON::GetRawStringValueFromJsonNode(iconNode, JSON::GetUtilityString(IconFileType)).value_or(""));

```

Comment:
Type was also required in the YAML; shouldn't this result in a "if (empty) { continue; }"?